src/deploy.rs

```diff
@@ -0,0 +1,427 @@
+// Copyright (c) 2023, MaidSafe.
+// All rights reserved.
+//
+// This SAFE Network Software is licensed under the BSD-3-Clause license.
+// Please see the LICENSE file for more details.
+
+use crate::{
+    error::{Error, Result},
+    print_duration, SnCodebaseType, TestnetDeploy,
+};
+use std::{net::SocketAddr, path::PathBuf, time::Instant};
+
+pub struct DeployCmd {
+    testnet_deploy: TestnetDeploy,
+    name: String,
+    node_count: u16,
+    vm_count: u16,
+    logstash_details: (String, Vec<SocketAddr>),
+    sn_codebase_type: SnCodebaseType,
+}
+
+impl DeployCmd {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        testnet_deploy: TestnetDeploy,
+        name: String,
+        node_count: u16,
+        vm_count: u16,
+        logstash_details: (String, Vec<SocketAddr>),
+        sn_codebase_type: SnCodebaseType,
+    ) -> Self {
+        Self {
+            testnet_deploy,
+            name,
+            node_count,
+            vm_count,
+            logstash_details,
+            sn_codebase_type,
+        }
+    }
+
+    pub async fn deploy(self) -> Result<()> {
```

Comment:
Minor style suggestion: if the class is a command, the term `execute` is often associated with running the command.

https://en.wikipedia.org/wiki/Command_pattern

The following:
```
let cmd = DeployCmd { };
cmd.execute();
```
Reads a little clearer than:
```
let cmd = DeployCmd { };
cmd.deploy();
```

Reply:
That is indeed cleaner! Thanks.